synapse/lib/types.py

```diff
@@ -552,6 +552,7 @@ def __init__(self):
         self.types = {}
         self.subtypes = []
 
+    def addDefaultTypes(self):
```

Comment:
i've pulled the basic types out of the typelib constructor.

this is so the cortex has a chance to persist the types. today, the typelib ctor is run before the cortex ctor, so the cortex does not have its storage setup yet. by splitting out this routine, we can add the basic types once we know the storage backend is ready.

conceptually, the typelib probably should function correctly with entirely use-provided type classes, giving a chance for them to install their own `str` type is potentially better.

unfortunately, this means there's an additional initialization call for any client that wants to use a typelib with the default types. today, this is just the cortex, and a bunch of the test cases.